clarity-ext-scripts/clarity_ext_scripts/covid/rtpcr_analyse_script.py

```diff
@@ -1,60 +1,64 @@
 from clarity_ext.extensions import GeneralExtension
 from clarity_ext_scripts.covid.rtpcr_analysis_service import RTPCRAnalysisService
+from clarity_ext_scripts.covid.rtpcr_analysis_service import MultipleAnalysisErrors
+from clarity_ext_scripts.covid.rtpcr_analysis_service import DIAGNOSIS_RESULT
+from clarity_ext_scripts.covid.rtpcr_analysis_service import COVID_RESPONSE_FAILED
+from clarity_ext_scripts.covid.rtpcr_analysis_service import CTResult
 from clarity_ext.domain.validation import UsageError
 
-BOUNDARY_DICT = {'Assay 1': (10, 50)}
-RT_PCR_POSITIVE_CONTROL = 'rtpcr_pos'
-RT_PCR_NEGATIVE_CONTROL = 'rtpcr_neg'
+
+RT_PCR_POSITIVE_CONTROL = 'Positive PCR Control'
+RT_PCR_NEGATIVE_CONTROL = 'Negative PCR Control'
 
 
 class Extension(GeneralExtension):
     def execute(self):
-        self.initial_validation()
-        ct_analysis_service = RTPCRAnalysisService()
-        lower_bound, upper_bound = BOUNDARY_DICT[self.assay]
-
-        rt_pcr_control_types = [RT_PCR_POSITIVE_CONTROL, RT_PCR_NEGATIVE_CONTROL]
-        found_controls = list()
-        self.context.current_step.udf_map.force("rtPCR Passed", False)
-        for _, output in self.context.all_analytes:
-            original_sample = output.sample()
-            original_sample.udf_map.force("rtPCR Passed", False)
+        if not self._has_assay_udf():
+            raise UsageError("The udf 'Assay' must be filled in before running this script")
 
-        # 1. Check control values
+        # Prepare analyse service input args
+        samples = list()
+        positive_controls = list()
+        negative_controls = list()
         for _, output in self.context.all_analytes:
-            original_sample = output.sample()
-            if original_sample.udf_control == 'Yes' \
-                    and original_sample.udf_control_type.lower() in rt_pcr_control_types:
-                found_controls.append(original_sample.udf_control_type.lower())
-                ct_analysis_service.validate_control_value(
-                    original_sample.udf_control_type, output.udf_ct,
-                    lower_bound, upper_bound)
+            # TODO: currently we only have udf for CT (no distinction between FAM and HEX)
```

Comment:
Yes, I agree here. We need more than one UDF. We should run it by Maike though, because we need to figure out if two is enough or if something like covid reporter (FAM) and internal control reporter (HEX/VIC) is alright for now.